### PR TITLE
Add python benchmark

### PR DESCRIPTION
**Pull request by pradithya**
Add serialization and deserialization benchmark for UPI and compare against JSON.
The benchmark is performed with various payload size and is controlled via 2 parameters (`n_rows` and `n_cols`) for both request and response payload. 

NOTE: the benchmark result can be seen in the test log. 
Screenshot 2022-09-01 at 12 41 04 PM

### CODE REVIEW

File: gen/python/grpc/test/benchmark_test.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+bench_response_test_cases = [
+    (1, 1),
+    (1, 10),
+    (100, 1),
+    (100, 10),
+    (1000, 1),
+    (1000, 10),
+]
+
+# represents number of rows x number of featuress
```
Comment (leonlnj) [inline gen/python/grpc/test/benchmark_test.py:21]: small typo

File: gen/python/grpc/test/benchmark_test.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    (1000, 500),
+]
+
+
+@pytest.mark.parametrize("n_rows,n_cols", bench_request_test_cases)
+@pytest.mark.benchmark(group="serialize-request")
+def test_serialize_proto_request(n_rows, n_cols, benchmark):
+    request = create_request(n_rows, n_cols, )
+    benchmark(serialize_request, request=request)
+    benchmark.extra_info["message_size"] = request.ByteSize()
```
Comment (leonlnj) [inline gen/python/grpc/test/benchmark_test.py:37]: where can this message size be found? cant find it anywhere

**Review by leonlnj — APPROVED** [review]
LGTM, seems like proto is generally faster. The catch is that, if this is done once/same number of times as http. 

Thanks!

File: gen/python/grpc/test/benchmark_test.py (hunk 2 above)
Comment (pradithya) [reply gen/python/grpc/test/benchmark_test.py:37]: You can only access it in the json report generated by the test. It's limitation of pytest-benchmark. I am exploring how can I expose it without having developer to clone and run the benchmark themselves.

**Pull request merged**